### ISSUE TITLE
Support for the Account MFE

### ISSUE THREAD

**Issue by arbrandes** [openedx/build-test-release-wg#27]
Coordinate, and if necessary execute, the tasks necessary to get the  [Account MFE](https://github.com/edx/frontend-app-account) up to speed for Lilac:

1. It passes a minimum standard of documentation:  at the very least, the READMEs should have descriptive text that includes use cases, screenshots, and particularly, documentation on that MFE's environment variables.
2. Its basic feature-set (as documented above) works reliably with the rest of the Lilac codebase. 
3. It's deployed by default using the Native Installation, and deployable as a plugin using Tutor.
4. It is internationalized (i18n), localizable (l10n), and passes minimum accessibility (a11y) standards.
5. It is reasonably themable.

**Comment by arbrandes** [comment]
@sarina, you ok with being assigned to help with item 1, above?

**Comment by sarina** [comment]
Yes, that's fine @arbrandes

**Comment by jfavellar90** [comment]
@arbrandes @BbrSofiane I'm looking to this task with my teammates and we have some comments on the points:

1. @sarina is there any ongoing effort on this matter from the edX team?
2. Our plan for this is to create a new native installation with the latest master version, and the install the account MFE to test it based on documentation in point 1.
3. We can work on deploying this MFE for the native installation by default, but we'll need some help with the tutor side where indeed there's an ongoing discussion [here](https://discuss.openedx.org/t/deploying-microfrontends-mfes-in-a-single-container/4649). During the following 2 weeks we cannot be too much involved on that thread. Are MFE's in tutor high priority in the short term?
4. From what we saw in the repo, there are i18n and l10n. Then we have the tests to spot possible translation errors. @sarina do you know if this info (i18n, l10n and a11y) is going to be included in the docs?
5. We need to investigate about theming MFE's. If ti's already clear for you guys, we really appreciate if you have docs or sources to look at.

We remain attentive to your feedback

**Comment by sarina** [comment]
@jfavellar90 - I have made a template for a good README (https://github.com/edx/frontend-template-application/blob/master/README-template-frontend-app.rst), and I'm working on cleaning up the installation instructions this week. The other parts of the template will be filled in by David Joy, who is currently out on vacation through next Wednesday.

**Comment by BbrSofiane** [comment]
@jfavellar90 regarding MFE theming, my understanding is that the documentation is scattered across different locations.

#36 - was created to address that and create a basic guide for theming. If the sources are provided is that something you could take on?

**Comment by sarina** [comment]
I've made a PR for the Account MFE README that I welcome feedback on. Note I tested the installation instructions on devstack and they worked, but there's a number of TODOs I'm waiting for David's return to address. https://github.com/edx/frontend-app-account/pull/427

**Comment by jfavellar90** [comment]
@sarina thanks, we'll take a look at that one. @BbrSofiane with this TODO we'll create a document with the followed procedure so it can be a base for #36

**Comment by sarina** [comment]
@arbrandes @jfavellar90 could you please review https://github.com/edx/frontend-app-account/pull/427 - it is ready to go from our end, but I want to make sure it meets your needs.

**Comment by arbrandes** [comment]
@jfavellar90, regarding Tutor, I think that for now we can rely on @regisb to lead the way.

Thanks for moving forward with this!

**Comment by regisb** [comment]
> regarding Tutor, I think that for now we can rely on @regisb to lead the way.

This is correct.

**Comment by BbrSofiane** [comment]
Hey @jfavellar90 any updates on the testing of the account MFE?

**Comment by jfavellar90** [comment]
@BbrSofiane hey, let me do some comments regarding the progress on this matter:
1. Since the account MFE is going to be the default in Lilac, we are proposing this change to the edx-platform repo: https://github.com/eduNEXT/edx-platform/commit/3220baa9dfa52c06b9810ff295a9516d0140188e where basically we are creating a new Toggle setting to enable or disable the account MFE globally (configuration_helpers requires the creation of a Django site). By default, this toggle would be True so the MFE is activated out of the box in a native installation. Notice that this change is deleting the waffle flag needed now in master to activate the MFE. Who's the right one to discuss this change and check the viability of this change to get merged?
2. This MFE gets deployed automatically with the native installation through the mfe_deployer role, but it will require some configuration variables to work properly (refer to https://github.com/edx/configuration/tree/open-release/lilac.master/playbooks/roles/mfe_deployer#comprehensive-example-of-a-deployment-using-subdirectories). Do we have to change the edx configuration edxapp role to add the defaults required to get the account MFE working in a fresh native installation? or maybe it's better doing the suggestion to add those variables with the right values in the config.yml before starting the native installation? For instance, by default ENABLE_CORS_HEADERS is disabled in configuration repo: https://github.com/edx/configuration/blob/open-release/lilac.master/playbooks/roles/edxapp/defaults/main.yml#L1025
3. I adhere to @regisb request regarding MFE env variables documentation: https://discuss.openedx.org/t/we-really-really-need-some-docs-on-mfe-env-settings/4788. We were trying with some values and after some try and failure with worked. We'll try to document the ones we configured.
4. I'll share credentials for you in the next few hours in a Lilac instance we are using for testing purposes.

cc @arbrandes

**Comment by regisb** [comment]
> By default, this toggle would be True so the MFE is activated out of the box in a native installation. Notice that this change is deleting the waffle flag needed now in master to activate the MFE. 

I :+1: this. Modifying the site configuration is a bit tricky to do in an automated manner.

> We were trying with some values and after some try and failure with worked.

You might want to have a look at my work-in-progress: https://github.com/overhangio/tutor-mfe/pull/1
I managed to get the account, profile and gradebook MFEs to work. Be careful with the profile url setting: https://github.com/overhangio/tutor-mfe/pull/1/files#diff-aa8f1f0b54e1304cb260e64d836458a57ee3164f2f1b284c92dddd419d48c2fcR8
(notice how it ends with "/u/").

**Comment by jfavellar90** [comment]
@BbrSofiane we've created a PR in the configuration repo: https://github.com/edx/configuration/pull/6381 to apply some fixes on the MFE deployment using the native installation. We added some variables to the openedx_native so the MFE's are almost ready to use (only Django admin configurations would be required to activate them). Would you mind taking a look at it? or who would be the right one for a review?

On the other hand, we have a more controversial change in mind which is to remove the Waffle flag for the account MFE activation and replace it with a SettingToggle (diff in https://github.com/eduNEXT/edx-platform/commit/3220baa9dfa52c06b9810ff295a9516d0140188e). I think this is a more delicate one that can be discussed once the first Lilac candidate is released.

**Comment by jfavellar90** [comment]
Hey @sarina  , do you know if it's possible to change some messages from the User Retirement section in the account MFE? These messages contain "edX"
![Screenshot from 2021-04-22 19-45-21](https://user-images.githubusercontent.com/15217219/115802311-42772280-a3a4-11eb-92c9-6ceb22528837.png)
Let us know if we can collaborate with the effort

**Comment by sarina** [comment]
I'm certain it's possible, I don't know how at the moment.

**Comment by BbrSofiane** [comment]
@jfavellar90 
- Did you manage to speak to someone at edX regarding the changes to the flags? Or do you need help on with this?
- Can you address @davidjoy comment on the PR to the configuration repo? 

@sarina @davidjoy what's the best way to go about making the changes for the text?

**Comment by sarina** [comment]
We have a ticket for it here - https://openedx.atlassian.net/browse/CENG-13 - you are welcome to pick it up, I keep saying I'm going to get to it "today" and that keeps not happening 😞 😭

**Comment by jfavellar90** [comment]
@BbrSofiane 
- I proceeded by creating this PR https://github.com/edx/edx-platform/pull/27518. I asked for a quick review to @regisb but it would be great if we can involve more people in this conversation (Could you help me with it?) 
- David's comment was already addressed. I'll apply a small change to get the PR ready.
- @sarina I'm unable to see the ticket. Do I need some special permissions?

**Comment by sarina** [comment]
@jfavellar90 yeah, we know 😞  we're working on it!

**Comment by jfavellar90** [comment]
@BbrSofiane @arbrandes this is the current state of this issue:

- The configuration PR https://github.com/edx/configuration/pull/6381 was merged. The native Lilac master was tested, and the MFE is working properly. Still, some messages need to be fixed. I'll create the PR to master in the next hours.
- There is a PR that is being discussed in https://github.com/edx/edx-platform/pull/27518. Basically, it changes the way the account MFE is activated. We were advocating to change the waffle flag by a regular feature flag in Django settings so you can control the activation via Ansible/Tutor settings globally or granularly with the site configuration model. We can discuss this a bit further at the next meeting (or visit the PR, your opinion is really appreciated). But I don't think this is a blocker for the first release candidate. The activation will still work via waffle flag and Site configuration.

**Comment by jfavellar90** [comment]
@BbrSofiane @arbrandes hey,

As we agreed in the BTR meeting, we've created the PR's in edx-platform and configuration:
https://github.com/edx/edx-platform/pull/27518
https://github.com/edx/configuration/pull/6402

The edx-platform PR is about to be merged, so once it lands in master, we'll proceed to backport to lilac master branch.
The configuration PR is just to create the Waffle flags. Normally when waffle flags need to be created on installation, they are added in Django data migrations (see an example [here](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/migrations/0018_add_waffle_flag_defaults.py) ), but knowing these flags are just temporary, we opted by adding the flags via Ansible as we discussed. Already tagged David Joy on that PR, but if you wanna take a look, go ahead.

Please let me know if you have any feedback/comment/complaint

**Comment by jfavellar90** [comment]
@BbrSofiane @arbrandes some updates on the PR's

1. The edx-platform PR for the account MFE waffle flag was merged: https://github.com/edx/edx-platform/pull/27518
We created a new PR in https://github.com/edx/edx-platform/pull/27623 to backport those changes to lilac master. Is the PR necessary? or someone with the right privileges can just cherry-pick the changes to Lilac master?

2. Regarding the configuration PR to create the MFE waffle flags:
We'd created this PR: https://github.com/edx/configuration/pull/6402 but it was created on Lilac master so we created a new PR https://github.com/edx/configuration/pull/6406 on top of master (as @nedbat suggested). I can close 6402 PR if this is not necessary.

3. Final point: the first PR we created in configuration for this MFE was merged on lilac.master: https://github.com/edx/configuration/pull/6381. I created a PR to port this change to master: https://github.com/edx/configuration/pull/6405. Could you please help me with reviewers to this one?

**Comment by BbrSofiane** [comment]
@jfavellar90 are the changes in edx/configuration#6406 backported to the lilac branch? Just asking to make sure I understood your comment. 

If we can backport edx/configuration#6406 to lilac.master and get edx/edx-platform#27623 merged then we should be ready to create a release candidate with the account MFE installed by default. Is that correct?

**Comment by jfavellar90** [comment]
@BbrSofiane https://github.com/edx/configuration/pull/6406 was already backported to Lilac master by @nedbat so I closed https://github.com/edx/configuration/pull/6402. 

The edx-platform change in https://github.com/edx/edx-platform/pull/27623 is still pending to be backported. Once this is backported, we are ready to go with the account MFE

**Comment by ericfab179** [comment]
@jfavellar90 @BbrSofiane Last edx-platform PR has already been merged. We launched a native installation using lilac.master and the redirection to the Account MFE is happening by default as expected.

**Comment by arbrandes** [comment]
I can confirm that the Account MFE is enabled by default on a new native install.  We can call this done - at least as far as edx/configuration is concerned.

@regisb, what's the status of the Tutor version of the Account MFE?

**Comment by regisb** [comment]
The Account MFE is working in Tutor as part of the lilac branch of the tutor-mfe plugin: https://github.com/overhangio/tutor-mfe/tree/lilac
